macOS (El Capitan): /karaf-edi/pom.xml – /bin/tar vs. /usr/bin/tar

Hi!

Building the project the first time fails on `karaf-edi`.
Here's the extended Log:

```shell
➜  mescedia-edi-server git:(master) mvn clean install -e
[INFO] Error stacktraces are turned on.
[INFO] Scanning for projects...
[WARNING]
[WARNING] Some problems were encountered while building the effective model for org.mescedia.edi:karaf-edi:pom:1.0
[WARNING] 'build.plugins.plugin.(groupId:artifactId)' must be unique but found duplicate declaration of plugin org.codehaus.mojo:exec-maven-plugin @ line 757, column 33
[WARNING]
[WARNING] It is highly recommended to fix these problems because they threaten the stability of your build.
[WARNING]
[WARNING] For this reason, future Maven versions might no longer support building such malformed projects.
[WARNING]
[INFO] ------------------------------------------------------------------------
[INFO] Reactor Build Order:
[INFO]
[INFO] mescedia :: Open Source EDI server [branding]
[INFO] mescedia :: Open Source EDI server [converter]
[INFO] mescedia :: Open Source EDI server [apache-karaf]
[INFO] mescedia :: Open Source EDI server [install]
[INFO]
[INFO] ------------------------------------------------------------------------
[INFO] Building mescedia :: Open Source EDI server [branding] 1.1.4
[INFO] ------------------------------------------------------------------------
[INFO]
[INFO] --- maven-clean-plugin:2.5:clean (default-clean) @ branding ---
[INFO] Deleting /Users/MyHomeFolder/mescedia-edi-server/karaf-branding/target
[INFO]
[INFO] --- maven-resources-plugin:3.0.2:resources (default-resources) @ branding ---
[WARNING] Using platform encoding (UTF-8 actually) to copy filtered resources, i.e. build is platform dependent!
[INFO] Copying 1 resource
[INFO]
[INFO] --- maven-compiler-plugin:3.6.1:compile (default-compile) @ branding ---
[INFO] No sources to compile
[INFO]
[INFO] --- maven-resources-plugin:3.0.2:testResources (default-testResources) @ branding ---
[WARNING] Using platform encoding (UTF-8 actually) to copy filtered resources, i.e. build is platform dependent!
[INFO] skip non existing resourceDirectory /Users/MyHomeFolder/mescedia-edi-server/karaf-branding/src/test/resources
[INFO]
[INFO] --- maven-compiler-plugin:3.6.1:testCompile (default-testCompile) @ branding ---
[INFO] No sources to compile
[INFO]
[INFO] --- maven-surefire-plugin:2.19.1:test (default-test) @ branding ---
[INFO] No tests to run.
[INFO]
[INFO] --- maven-bundle-plugin:2.2.0:bundle (default-bundle) @ branding ---
[INFO]
[INFO] --- maven-install-plugin:2.5.2:install (default-install) @ branding ---
[INFO] Installing /Users/MyHomeFolder/mescedia-edi-server/karaf-branding/target/branding-1.1.4.jar to /Users/MyHomeFolder/.m2/repository/org/mescedia/edi/branding/1.1.4/branding-1.1.4.jar
[INFO] Installing /Users/MyHomeFolder/mescedia-edi-server/karaf-branding/pom.xml to /Users/MyHomeFolder/.m2/repository/org/mescedia/edi/branding/1.1.4/branding-1.1.4.pom
[INFO]
[INFO] --- maven-bundle-plugin:2.2.0:install (default-install) @ branding ---
[INFO] Installing org/mescedia/edi/branding/1.1.4/branding-1.1.4.jar
[INFO] Writing OBR metadata
[INFO]
[INFO] ------------------------------------------------------------------------
[INFO] Building mescedia :: Open Source EDI server [converter] 1.1.4
[INFO] ------------------------------------------------------------------------
[INFO]
[INFO] --- maven-clean-plugin:2.5:clean (default-clean) @ converter ---
[INFO] Deleting /Users/MyHomeFolder/mescedia-edi-server/unedifact-converter/target
[INFO]
[INFO] --- maven-resources-plugin:2.6:resources (default-resources) @ converter ---
[INFO] Using 'UTF-8' encoding to copy filtered resources.
[INFO] Copying 0 resource
[INFO]
[INFO] --- maven-compiler-plugin:2.5.1:compile (default-compile) @ converter ---
[INFO] Compiling 15 source files to /Users/MyHomeFolder/mescedia-edi-server/unedifact-converter/target/classes
[INFO]
[INFO] --- maven-resources-plugin:2.6:testResources (default-testResources) @ converter ---
[INFO] Using 'UTF-8' encoding to copy filtered resources.
[INFO] skip non existing resourceDirectory /Users/MyHomeFolder/mescedia-edi-server/unedifact-converter/src/test/resources
[INFO]
[INFO] --- maven-compiler-plugin:2.5.1:testCompile (default-testCompile) @ converter ---
[INFO] No sources to compile
[INFO]
[INFO] --- maven-surefire-plugin:2.12.4:test (default-test) @ converter ---
[INFO] No tests to run.
[INFO]
[INFO] --- maven-jar-plugin:2.4:jar (default-jar) @ converter ---
[INFO] Building jar: /Users/MyHomeFolder/mescedia-edi-server/unedifact-converter/target/converter-1.1.4.jar
[INFO]
[INFO] --- maven-install-plugin:2.4:install (default-install) @ converter ---
[INFO] Installing /Users/MyHomeFolder/mescedia-edi-server/unedifact-converter/target/converter-1.1.4.jar to /Users/MyHomeFolder/.m2/repository/org/mescedia/unedifact/converter/1.1.4/converter-1.1.4.jar
[INFO] Installing /Users/MyHomeFolder/mescedia-edi-server/unedifact-converter/pom.xml to /Users/MyHomeFolder/.m2/repository/org/mescedia/unedifact/converter/1.1.4/converter-1.1.4.pom
[INFO]
[INFO] ------------------------------------------------------------------------
[INFO] Building mescedia :: Open Source EDI server [apache-karaf] 1.0
[INFO] ------------------------------------------------------------------------
[INFO]
[INFO] --- maven-clean-plugin:2.5:clean (default-clean) @ karaf-edi ---
[INFO] Deleting /Users/MyHomeFolder/mescedia-edi-server/karaf-edi/target
[INFO]
[INFO] --- maven-dependency-plugin:2.8:copy (get karaf archive and libs) @ karaf-edi ---
[INFO] Configured Artifact: org.apache.karaf:apache-karaf:4.0.8:tar.gz
[INFO] Configured Artifact: org.mescedia.edi:branding:1.1.4:jar
[INFO] Configured Artifact: mysql:mysql-connector-java:5.1.34:jar
[INFO] Configured Artifact: org.milyn:milyn-smooks-all:1.6:jar
[INFO] Copying apache-karaf-4.0.8.tar.gz to /Users/MyHomeFolder/mescedia-edi-server/karaf-edi/target/dependencies/apache-karaf-4.0.8.tar.gz
[INFO] Copying branding-1.1.4.jar to /Users/MyHomeFolder/mescedia-edi-server/karaf-edi/target/dependencies/apache-karaf-4.0.8/lib/ext/karaf-org.mescedia.edi.branding-1.1.4.jar
[INFO] Copying mysql-connector-java-5.1.34.jar to /Users/MyHomeFolder/mescedia-edi-server/karaf-edi/target/dependencies/apache-karaf-4.0.8/lib/ext/karaf-mysql-connector-java-5.1.34.jar
[INFO] Copying milyn-smooks-all-1.6.jar to /Users/MyHomeFolder/mescedia-edi-server/karaf-edi/target/dependencies/apache-karaf-4.0.8/lib/ext/karaf-milyn-smooks-all-1.6.jar
[INFO]
[INFO] --- exec-maven-plugin:1.4.0:exec (extract karaf archive) @ karaf-edi ---
[INFO] ------------------------------------------------------------------------
[INFO] Reactor Summary:
[INFO]
[INFO] mescedia :: Open Source EDI server [branding] ...... SUCCESS [  0.866 s]
[INFO] mescedia :: Open Source EDI server [converter] ..... SUCCESS [  1.808 s]
[INFO] mescedia :: Open Source EDI server [apache-karaf] .. FAILURE [  0.538 s]
[INFO] mescedia :: Open Source EDI server [install] ....... SKIPPED
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time: 3.540 s
[INFO] Finished at: 2017-03-09T22:03:15+01:00
[INFO] Final Memory: 62M/380M
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal org.codehaus.mojo:exec-maven-plugin:1.4.0:exec (extract karaf archive) on project karaf-edi: Command execution failed. Cannot run program "/bin/tar" (in directory "/Users/MyHomeFolder/mescedia-edi-server/karaf-edi"): error=2, No such file or directory -> [Help 1]
org.apache.maven.lifecycle.LifecycleExecutionException: Failed to execute goal org.codehaus.mojo:exec-maven-plugin:1.4.0:exec (extract karaf archive) on project karaf-edi: Command execution failed.
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:212)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:153)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:145)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:116)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:80)
    at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build(SingleThreadedBuilder.java:51)
    at org.apache.maven.lifecycle.internal.LifecycleStarter.execute(LifecycleStarter.java:128)
    at org.apache.maven.DefaultMaven.doExecute(DefaultMaven.java:307)
    at org.apache.maven.DefaultMaven.doExecute(DefaultMaven.java:193)
    at org.apache.maven.DefaultMaven.execute(DefaultMaven.java:106)
    at org.apache.maven.cli.MavenCli.execute(MavenCli.java:863)
    at org.apache.maven.cli.MavenCli.doMain(MavenCli.java:288)
    at org.apache.maven.cli.MavenCli.main(MavenCli.java:199)
    at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
    at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
    at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke(Method.java:498)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced(Launcher.java:289)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launch(Launcher.java:229)
    at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode(Launcher.java:415)
    at org.codehaus.plexus.classworlds.launcher.Launcher.main(Launcher.java:356)
Caused by: org.apache.maven.plugin.MojoExecutionException: Command execution failed.
    at org.codehaus.mojo.exec.ExecMojo.execute(ExecMojo.java:281)
    at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo(DefaultBuildPluginManager.java:134)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:207)
    ... 20 more
Caused by: java.io.IOException: Cannot run program "/bin/tar" (in directory "/Users/MyHomeFolder/mescedia-edi-server/karaf-edi"): error=2, No such file or directory
    at java.lang.ProcessBuilder.start(ProcessBuilder.java:1048)
    at java.lang.Runtime.exec(Runtime.java:620)
    at org.apache.commons.exec.launcher.Java13CommandLauncher.exec(Java13CommandLauncher.java:61)
    at org.apache.commons.exec.DefaultExecutor.launch(DefaultExecutor.java:279)
    at org.apache.commons.exec.DefaultExecutor.executeInternal(DefaultExecutor.java:336)
    at org.apache.commons.exec.DefaultExecutor.execute(DefaultExecutor.java:166)
    at org.codehaus.mojo.exec.ExecMojo.executeCommandLine(ExecMojo.java:660)
    at org.codehaus.mojo.exec.ExecMojo.execute(ExecMojo.java:265)
    ... 22 more
Caused by: java.io.IOException: error=2, No such file or directory
    at java.lang.UNIXProcess.forkAndExec(Native Method)
    at java.lang.UNIXProcess.<init>(UNIXProcess.java:247)
    at java.lang.ProcessImpl.start(ProcessImpl.java:134)
    at java.lang.ProcessBuilder.start(ProcessBuilder.java:1029)
    ... 29 more
[ERROR]
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR]
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoExecutionException
[ERROR]
[ERROR] After correcting the problems, you can resume the build with the command
[ERROR]   mvn <goals> -rf :karaf-edi
```

Thanks in advance for your feedback!

--- a/karaf-edi/pom.xml
+++ b/karaf-edi/pom.xml
@@ -28,21 +28,24 @@
 	<url>https://www.mescedia.com</url>
 	<name>mescedia :: Open Source EDI server [apache-karaf] </name>
 	
+	<properties>
+		<karaf.version>4.0.8</karaf.version>
+  		<mysql.version>5.1.34</mysql.version>
+		<smooks.version>1.6</smooks.version>
+  		<hawtio.version>1.4.66</hawtio.version>
+		<camel.version>2.16.4</camel.version>
+  		<activemq.version>5.14.1</activemq.version>	
+		<install.target>${basedir}/target/dependencies</install.target>
+		<install.archive>apache-karaf-${karaf.version}.tar.gz</install.archive>  		
+	</properties>
+	
 	<profiles>
   		<profile>
   			<id>default</id>
   			<activation>
-      			<activeByDefault>true</activeByDefault>
-    		</activation>
+      				<activeByDefault>true</activeByDefault>
+    			</activation>
 		  	<properties>
-		  		<karaf.version>4.0.8</karaf.version>
-		  		<mysql.version>5.1.34</mysql.version>
-		  		<smooks.version>1.6</smooks.version>
-		  		<hawtio.version>1.4.66</hawtio.version>
-		  		<camel.version>2.16.4</camel.version>
-		  		<activemq.version>5.14.1</activemq.version>		
-		  		<install.target>${basedir}/target/dependencies</install.target>
-		  		<install.archive>apache-karaf-${karaf.version}.tar.gz</install.archive>
 		  		<exec.path>/bin</exec.path>
 		  	</properties>
 	  	</profile>
@@ -50,14 +53,6 @@
 	  	<profile>
 	  		<id>macOS</id>
 		  	<properties>
-		  		<karaf.version>4.0.8</karaf.version>
-		  		<mysql.version>5.1.34</mysql.version>
-		  		<smooks.version>1.6</smooks.version>
-		  		<hawtio.version>1.4.66</hawtio.version>
-		  		<camel.version>2.16.4</camel.version>
-		  		<activemq.version>5.14.1</activemq.version>		
-		  		<install.target>${basedir}/target/dependencies</install.target>
-		  		<install.archive>apache-karaf-${karaf.version}.tar.gz</install.archive>
 		  		<exec.path>/usr/bin</exec.path>
 		  	</properties>
 	  	</profile>

--- a/karaf-edi/pom.xml
+++ b/karaf-edi/pom.xml
@@ -28,16 +28,40 @@
 	<url>https://www.mescedia.com</url>
 	<name>mescedia :: Open Source EDI server [apache-karaf] </name>
 	
-	<properties>
-		<karaf.version>4.0.8</karaf.version>
-		<mysql.version>5.1.34</mysql.version>
-		<smooks.version>1.6</smooks.version>
-		<hawtio.version>1.4.66</hawtio.version>
-		<camel.version>2.16.4</camel.version>
-		<activemq.version>5.14.1</activemq.version>		
-		<install.target>${basedir}/target/dependencies</install.target>
-		<install.archive>apache-karaf-${karaf.version}.tar.gz</install.archive>
-	</properties>
+	<profiles>
+  		<profile>
+  			<id>default</id>
+  			<activation>
+      			<activeByDefault>true</activeByDefault>
+    		</activation>
+		  	<properties>
+		  		<karaf.version>4.0.8</karaf.version>
+		  		<mysql.version>5.1.34</mysql.version>
+		  		<smooks.version>1.6</smooks.version>
+		  		<hawtio.version>1.4.66</hawtio.version>
+		  		<camel.version>2.16.4</camel.version>
+		  		<activemq.version>5.14.1</activemq.version>		
+		  		<install.target>${basedir}/target/dependencies</install.target>
+		  		<install.archive>apache-karaf-${karaf.version}.tar.gz</install.archive>
+		  		<exec.path>/bin</exec.path>
+		  	</properties>
+	  	</profile>
+
+	  	<profile>
+	  		<id>macOS</id>
+		  	<properties>
+		  		<karaf.version>4.0.8</karaf.version>
+		  		<mysql.version>5.1.34</mysql.version>
+		  		<smooks.version>1.6</smooks.version>
+		  		<hawtio.version>1.4.66</hawtio.version>
+		  		<camel.version>2.16.4</camel.version>
+		  		<activemq.version>5.14.1</activemq.version>		
+		  		<install.target>${basedir}/target/dependencies</install.target>
+		  		<install.archive>apache-karaf-${karaf.version}.tar.gz</install.archive>
+		  		<exec.path>/usr/bin</exec.path>
+		  	</properties>
+	  	</profile>
+  	</profiles>
 		
 	<dependencies>
 		<dependency>
@@ -649,7 +673,7 @@
 					  <argument>-C</argument>
 					  <argument>${install.target}</argument>					  
 					</arguments>		  					
-					<executable>/bin/tar</executable>
+					<executable>${exec.path}/tar</executable>
 			      </configuration>
 			    </execution>				
 			  </executions>

